README.md
# PokerPro
一、登陆
1、手机+密码登录
2、手机+验证码登录
3、邮箱+密码登录

二、注册
1、手机验证码+密码注册
2、邮箱+密码注册

三、主页

1、主页界面展示
2、左侧滑菜单
3、左侧滑菜单-昵称栏可点->编辑资料：完成头像图片选择（相机、相册）、性别选择器、日期选择器
4、编辑资料功能完成，编辑资料后刷新主页
5、新增找回密码功能
6、新增设置UI、帐户安全UI
7、新增设置-修改密码功能
8、日历、md5、多国语言
9、新主页界面UI、即将到来的赛事列表
10、登陆UI改版
11、注册UI改版
12、忘记密码UI改版
13、赛事详情静态数据
14、发送邮箱验证码
15、赛事详情购票跳转到购票UI（购票界面暂无功能）
16、新增Splash启动页，消除启动时白板
17、购票页、购票须知、实名认证、电子购票

tony测试

change_log.txt
2018年4月11日-2018年4月23
1、完成附近的牌友
2、完成发现
3、完成广场关注动态
4、修复商场无图显示默认图，包邮
5、Android 聊天室适配
本周计划：
1、完成翻译工作
2、处理好Android视频接受问题
3、完成发布社交工作


2018年3月26日-2018年4月1
1、完成产品故事AC编写，AC总计130，第一期迭代AC58
2、集成极光IMSDK，调试AndroidIM界面和功能
3、首页发布界面编写完成



2018年3月19日-2018年3月26
1、打包发布本地化IOS，解决其他国家下载不了，新增了韩国、日本、越南、马来西亚、英语
2、React Native升级到最新，适配升级后UI适配
3、替换Umeng服务，转使用极光
本周计划：
1、测试升级React Native后app的回归测试
2、开发社交IM、

2018年3月11日-2月24日
1、升级React Native在Debug模式下正常，Release模式闪退
2、完成扑客币对购票、众筹、商城的使用

2018年2月25日-3月2日
1、调整部分UI
2、发布IOS和android 版本3.3
3、招聘React Native
4、已升级React native版本，但是部分第三方插件还需跟新（jpush、Umeng统计、Umeng分享、微信）



2018年2月11日-2月24日
1、优化中英文切换，热点资讯和视频列表刷新
2、UI界面调整
3、资讯、视频。内容链接跳转App内部界面（赛事、资讯、视频、众筹、商品）
4、修复android首页卡屏


##### [跳App赛事](approuter://race/252)

##### [跳App商品](approuter://mall/24)

##### [跳App资讯](approuter://news/73)

##### [跳App视频](approuter://video/14)

##### [跳App众筹](approuter://crowd/8)

2018年2月1日-2月11日
本次重要更新：
1、首页热门资讯分页加载
2、众筹赞助（列表、详情、牌手列表、牌手详情、及时赛报、牌手赛报、认购、支付、订单列表、订单详情）
3、登录优化（比如在未登录时，在购票页，需登录，旧版登录跳转主页，调整后默认回原来的界面）


2017年12月21日-2017年12月30日 IOS & Android
1、完成上一次App迭代版本的测试和发布，IOS和Android都已上线
2、完成官网的首页界面
3、编写众筹AC编写
4、完成众筹倒计时控件和众筹认购进度条控件

2017年12月30日-2017年12月14日 IOS & Android
1、在没有设计稿开发众筹首页和详情
2、开发众筹相关界面



2017年11月30日-2017年12月13日 IOS & Android
1、完成迭代所有开发内容
2、完成评论删除
3、支付限额提示
4、官网开发



2017年11月12日-2017年11月26日 IOS & Android
1、H5与App通信，H5调用原生的方法，原生调用H5的方法。
2、完成点赞
3、完成H5评论列表分页加载
4、写评论、写回复
5、完成评论详情
6、资讯、视频嵌入H5



2017年10月30日-2017年11月8日 IOS & Android
1、完成商城首页（类别分页、商品子目录、子目录商品)
2、商品搜索（历史记录保存、搜索分页、搜索加载更多）
3、修复排行UI适配问题
4、首页加载不出，提示重新加载

2017年10月23日-2017年10月28日 IOS & Android
1、申请IOS企业开发证书（邓白氏吗申请提交->以得到邓白氏码->成功提交IOS企业证书申请)
2、完成首页商城UI：低部导航、商城类别、商城产品列表、商品排序、商品分类
3、完成商品搜索UI

2017年10月8日-2017年10月13日 IOS & Android
1、完成轮播图控件
2、首页资讯视频
3、首页排行
4、首页头条、火热开赛、即将到来、热门资讯（By 小米）
5、我的UI （By小米）
6、适配新首页的内部逻辑（登陆、登出）等状态控制
7、视频控件重构、播放界面调整
8、首页中英切换更新界面数据
9、视频播放暂停控制
10、首页返回顶部
11、邀请码折扣功能
12、视频界面改版完成
13、用户状态管理


2017年9月25日-2017年9月30日 IOS & Android
1、完成迭代内容
2、赛事详情UI更新，（动画效果，改善用户体验）
3、修复资讯UI适配
4、优化路由

2017年9月18日-2017年9月22日 IOS & Android
1、修复android 闪退bug
2、升级React Native
3、优化界面跳转
4、完成实名列表
5、完成实名添加
6、完成实名修改（待审核、审核失败）
7、完成实名查看（审核通过）
8、删除实名
9、未登录可进入选票查看，套票详情，分享套票


2017年9月11日-2017年9月8日 IOS & Android
1、修复个人中心日期bug
2、修复Umeng分享跟微信登陆冲突(android)

2017年9月4日-2017年9月8日 IOS & Android
1、推荐码
2、在选票界面如果已下单，直接跳转到待付款界面
3、完成消息中心界面
4、活动推送
5、活动消息
6、优化图文富文本控件
7、活动有H5跳WebView，否则跳本地
8、APP更新提示
9、未读消息提示
10、优化android 视频播放卡顿


2017年8月21日-2017年8月25日 IOS & Android
1、意见反馈
2、完成微信支付
3、赛票金额，有购票才显示
4、微信登陆、微信绑定
5、更多赛事，筛选
6、登陆UI界面改版



2017年8月14日-2017年8月18日 IOS & Android
1、优化购票流程，已下单未付款，再次购票是直接跳转到未付款
2、集成微信登陆，获取微信用户信息
3、集成微信支付
4、优化排名UI



2017年8月7日-2017年8月10日 IOS & Android
1、完成购票支付流程业务
2、完成订单支付流程业务
3、订单列表取消
4、订单详情界面调整
5、查看物流、确认收货
6、在付款界面金额确实、订单列表联系客服闪退、金额¥、UI微调



2017年7月31日-2017年8月4日 IOS & Android
1、修复上次迭代出现的9个问题
2、修复PokerPro测试问题汇总列表
3、新增视频列表点击播放
4、完成去支付UI界面



2017年7月24日-2017年7月29日 IOS & Android
1、App扑客修改方案 完成
2、中英文翻译更新
3、用户登陆统计
4、用户被封禁
5、排行-个人业务完成
6、排行-赛事业务完成
7、关注功能完成
8、排行列表
9、研究React Native 单元测试架构
10、排行首页过业务---完成
11、关注列表取消关注
13、牌手战绩--赛事标题-主赛-边塞
14、排行赛事-金额格式化
15、牌手分享、赛事结果分享


2017年7月17日-2017年7月21日 IOS & Android
1、资讯、视频标题导航栏UI调整，视频列表改版
2、设置中多个页面UI调整
3、个人资料UI调整，选择器蒙板等
4、修复已知bug
5、购票-键盘遮挡问题修复
6、接口中英文切换
7、赛程表格式 换换行"|"
8、更多赛事上拉修复优化
9、App中英翻译更新
10、App扑客修改方案（界面改造完成，数据调整正在跟进）

2017年7月10日-2017年7月14日 IOS & Android
1、收货地址列表侧滑默认、删除、编辑
2、字体大小调整
3、资讯、视频界面UI调整
4、修复迭代提出的5个问题
5、迭代完成验收后，产品确认收货，我这边会直接上传新版到AppStore，由周庆提交送审。（未确认）
6、图文显示控件，图片覆盖文字原因是：**（加粗）


2017年7月3日-2017年7月11日 IOS & Android
1、登陆记录账户
2、主页UI改版
3、赛事分享（标题、时间地点、图标）
4、语言切换
5、主页动画效果
6、主页赛事列表
7、更多赛事列表
8、更多赛事类型
9、赛事搜索UI改版
10、新增收货地址
11、编辑收货地址
12、收货地址列表（测滑设置默认和删除还没有实现）
13、购票-收货地址同步刷新
14、收货地址列表侧滑默认、删除、编辑
15、字体大小调整
16、资讯、视频界面UI调整


2017年6月26日-2017年6月30日 IOS & Android
1、配合maxUI界面调整
2、集成Umeng，IOS和Android SDK调试
3、调试Android端微信、QQ、微博
4、调试IOS端
5、修复边赛-边赛结果下滑不全
6、边赛起始筹码
7、微信、微博、QQ赛事信息分享完成


2017年6月19日-2017年6月23日 IOS & Android
1、购票须知客服热线
2、选购、购票查看详情，套票进入套票详情，仅赛票进入赛事详情
3、Umeng事件布点
4、修复视频正在加载中不能返回
5、重构选票接口和界面逻辑
6、重构订单接口和界面
7\lskjdflajsdklfj

2017年6月12日-2017年6月16日 IOS & Android
1、添加WebView界面功能，需UI设计
2、试发布IOS AppStore
3、App上线版本修改已完成50%


2017年6月5日-2017年6月9日 IOS & Android
1、修复上个迭代消息推送和视频播放控件的一些bug
2、优化资讯和视频下拉加载，但是去除了空界面提示（控件冲突）
3、选票界面按钮交互设计
4、赛事选票功能主赛和边赛
5、赛事选票有无，控制是否可以点击
6、选票边赛底部弹窗选择列表
7、选票主赛列表
8、首页"排行"，有选票主赛和边赛数据
9、选票-购票页，完成局部UI改版
10、完成购票接口更新
11、完成赛票详情页功能



2017年5月22日-2017年5月26日 IOS & Android
1、替换React Native壳
2、集成Jpush消息推送
3、收到推送未读消息角标和标红提示
4、视频播放控件,在首页点击视频
5、视频类型接口对接完成，UI还需调整
6、视频列表接口对接完成，UI还需调整
7、视频详情接口对接完成，UI还需调整
8、视频播发控制，暂停、全屏、拖动进度、播完点击重播，点击屏幕显示和隐藏控制栏
9、视频图片加载占位图和遮罩


2017年5月15日-2017年5月19日 IOS & Android
1、修复 更换手机号完成时跳转登录界面
2、修复 资讯内容滚动
3、修复 手机号更换，手动获取验证码

2017年5月8日-2017年5月12日 IOS & Android
1、完成主赛盲注结构业务
2、完成主赛赛程边业务
3、完成边赛盲注结构业务
4、完成赛程表业务
5、添加Api环境切换，切换后关闭App重启
6、完成票务列表业务
7、完成票务搜索业务
8、优化主赛、边赛信息
9、实现绑定手机UI
10、完成绑定手机业务
11、完成更换绑定手机业务（一个月更换一次还未实现）
12、完成绑定手机和更换手机AC
13、修复 1当退出登录时，在输入账户登录账号后，应该进入首页而不是个人中心页

-2017年5月5日 IOS & Android
1、修复迭代会上资讯更新问题
2、赛程表、盲注结构、赛事结果动态显示
3、研究React Native 0.44更新了Navigation,旧的Navigator被删除，下个迭代替换
4、完成盲注结构UI
5、完成赛程表UI



2017年4月24日-2017年4月28日 IOS & Android
1、初步完成资讯搜索业务
2、兼容首页资讯android新控件
3、首页资讯下拉刷新，上拉加载
4、重构Markdown，扩展换行，空格
5、在跳转赛事详情做了，跳转后渲染UI
6、更多赛事改造：主办方赛选列表
7、重构Markdown，IOS支持换行、空格，android 暂不支持
8、资讯完善
9、完成更多赛事主办方筛选业务
10、优化跳转赛事详情过程卡顿
11、研究更多赛事新控件
12、更多赛事下拉刷新后停留在原位置
13、优化跳转
14、完成资讯主页业务、资讯详情、资讯搜索
15、更多赛事主办方过滤业务、优化更多赛事结构
16、赛事详情根据是否有边赛显示边赛分页，是否有主赛结果显示主赛结果分页
17、研究视频播放器


2017年4月17日-2017年4月21日 IOS & Android
1、赛事详情去掉了，赛事简介（正在设计新样式）
2、新版赛事详情UI
3、界面UI微调
4、修复赛事排行缓存问题
5、初步完成资讯主页界面业务（导航、展示列表）
6、初步完成资讯详情
7、研究Markdown样式
8、界面调转卡顿尝试优化
9、自定义图片加载占位图


2017年4月10日-2017年4月14日 IOS & Android
1、完成边赛列表业务
2、主赛结果业务
3、边赛信息业务
4、边赛结果业务
5、参赛人信息业务
6、获取图片优化




2017年4月4日-2017年4月7日 IOS & Android
1、实名认证，只有审核通过了不允许编辑
2、更多赛事，添加"本月"功能
3、购票状态
4、购票页下拉刷新
5、赛事详情-边赛信息-主赛结果UI
6、赛事信息业务：目前只有"TRITON－传奇扑克超高额豪客赛2017巡回赛"有边赛信息


2017年3月27日-2017年3月31日 IOS & Android
1、实名信息审核失败，备注信息显示
2、赛事是否有购票
3、引入配置文件，用户配置管理第三发APP_KEY
4、完成日历有赛事标记提示
5、更新赛事接口，是否有票在售、是否有详情可点、实名审核失败提示
6、完成AC16、AC15
7、界面微调：首页、赛事详情、订单、购票等
8、修复已付款--为待发货
9、赛事详情，图片浏览左右滑动浏览多图
10、分享
11、头像裁剪
12、首页点击头像查看大图
13、短信验证




src/Themes/Fonts.js
// @flow
import StorageKey from '../configs/StorageKey';
const type = {
    base: 'PingFangSC-Light',
};

const size = {
    h1: 38,
    h36: 36,
    h30: 30,
    h26: 26,
    h24: 24,
    h17: 17,
    h18: 18,
    input: 18,
    h16: 16,
    h15: 15,
    h14: 14,
    h13: 13,
    h12: 12,
    h9: 9,
    tiny: 8.5
};


const style = {
    h1: {
        fontFamily: type.base,
        fontSize: size.h1
    },
    h36: {
        fontFamily: type.base,
        fontSize: size.h36
    },
    h30: {
        fontFamily: type.base,
        fontSize: size.h30
    },
    h26: {
        fontSize: size.h26
    },
    h24: {
        fontSize: size.h24
    },
    h18: {
        fontSize: size.h18
    },
    normal: {
        fontFamily: type.base,
        fontSize: size.h15
    },
    description: {
        fontFamily: type.base,
        fontSize: size.h14
    }
}

export default {
    H11: {
        fontFamily: type.base,
        fontSize: 11
    },
    H12: {
        fontFamily: type.base,
        fontSize: 12
    },
    H13: {
        fontFamily: type.base,
        fontSize: 13
    },
    H14: {
        fontFamily: type.base,
        fontSize: 14
    },
    H15: {
        fontFamily: type.base,
        fontSize: 15
    },
    H16: {
        fontFamily: type.base,
        fontSize: 16
    },
    H17: {
        fontFamily: type.base,
        fontSize: 17
    },
    H18: {
        fontFamily: type.base,
        fontSize: 18
    },
    H19: {
        fontFamily: type.base,
        fontSize: 19
    },
    H20: {
        fontFamily: type.base,
        fontSize: 20
    },
    H21: {
        fontFamily: type.base,
        fontSize: 21
    },
    type,
    size,
    style,
}

